src/index.ts

```diff
@@ -155,6 +157,46 @@ function endIndex(str: string, index: number, min: number) {
   return min;
 }
 
+export interface StringifyOptions {
+  /**
+   * Specifies a function that will be used to encode a [cookie-value](https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.1).
+   * Since value of a cookie has a limited character set (and must be a simple string), this function can be used to encode
+   * a value into a string suited for a cookie's value, and should mirror `decode` when parsing.
+   *
+   * @default encodeURIComponent
+   */
+  encode?: (str: string) => string;
+}
+
+/**
+ * Stringifies an object into a HTTP `Cookie` header.
+ */
+export function stringify(
+  cookies: Cookies,
+  options?: StringifyOptions,
+): string {
+  const enc = options?.encode || encodeURIComponent;
+  const cookieStrings: string[] = [];
+
+  for (const [name, val] of Object.entries(cookies)) {
+    if (val === undefined) continue;
+
+    if (!cookieNameRegExp.test(name)) {
+      throw new TypeError(`cookie name is invalid: ${name}`);
+    }
+
+    const value = enc(val);
+
+    if (!cookieValueRegExp.test(value)) {
+      throw new TypeError(`cookie val is invalid: ${val}`);
+    }
```

Comment:
Maybe this is more about serialize, but we end up dropping something with say `{foo: undefined}` here silently.

Whereas serialize will give you `foo=undefined`

is the discrepancy intended? v0 line also serializes undefined that same way

Reply:
> is the discrepancy intended?

Yes, it's intended. The design of `parse` allows for `undefined` to be returned and skip that entity when parsing, so this is mirroring support that (where an object can have `undefined` as the value).

As for the serialize API, it's documented to support `string` on `serialize` and that's the defined support, anything else is out of scope for that API. We shouldn't be trying to support non-strings even if they happen to work today accidentally.